portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/SkinnableCharonPortal.java

```diff
@@ -1735,13 +1735,18 @@ public void includeBottom(PortalRenderContext rcontext, Site site)
                         			log.error(e1.getMessage(), e1);
                         		}
                         	}
-						}
+                        }
 
                         if(sakaiThemeSwitcherEnabled) {
                             rcontext.put("themeSwitcher", true);
+                            
+                            String userTheme = "sakaiUserTheme-notSet";
+                            
+                            //ResourceProperties themePrefs = prefs.getProperties(org.sakaiproject.user.api.PreferencesService.USER_SELECTED_UI_THEME_PREFS);
+                            userTheme = prefs.getProperties(org.sakaiproject.user.api.PreferencesService.USER_SELECTED_UI_THEME_PREFS).getProperty("theme");
```

Comment:
```suggestion
                            String userTheme = StringUtils.defaultIfEmpty(prefs.getProperties(org.sakaiproject.user.api.PreferencesService.USER_SELECTED_UI_THEME_PREFS).getProperty("theme"), "sakaiUserTheme-notSet");

```

Reply:
Thanks for the suggestion. I've made this change.